benefits/enrollment_switchio/views.py

```diff
@@ -35,39 +40,24 @@ class GatewayUrlView(AgencySessionRequiredMixin, EligibleSessionRequiredMixin, V
     """View for the tokenization gateway registration"""
 
     def get(self, request: HttpRequest, *args, **kwargs):
-        switchio_config = self.agency.switchio_config
-
-        client = Client(
-            api_url=switchio_config.api_base_url,
-            api_key=switchio_config.api_key,
-            api_secret=switchio_config.api_secret,
-            private_key=switchio_config.private_key_data,
-            client_certificate=switchio_config.client_certificate_data,
-            ca_certificate=switchio_config.ca_certificate_data,
-        )
-
-        route = reverse(routes.ENROLLMENT_SWITCHIO_INDEX)
-        redirect_url = _generate_redirect_uri(request, route)
-
-        registration = client.request_registration(
-            eshopRedirectUrl=redirect_url,
-            mode=RegistrationMode.REGISTER,
-            eshopResponseMode=EshopResponseMode.FORM_POST,
-            timeout=settings.REQUESTS_TIMEOUT,
-        )
-        Session(request=request, registration_id=registration.regId)
+        response = request_registration(request, self.agency.switchio_config)
 
-        data = {"gateway_url": registration.gtwUrl}
-        return JsonResponse(data)
+        if response.status is Status.SUCCESS:
+            registration = response.registration
 
+            Session(request=request, registration_id=registration.regId)
 
-# copied from https://github.com/Office-of-Digital-Services/django-cdt-identity/blob/main/cdt_identity/views.py#L42-L50
-def _generate_redirect_uri(request: HttpRequest, redirect_path: str):
-    redirect_uri = str(request.build_absolute_uri(redirect_path)).lower()
+            data = {"gateway_url": registration.gtwUrl}
+            return JsonResponse(data)
+        else:
+            logger.debug("Error occurred while requesting access token", exc_info=response.exception)
```

Comment:
Do we want to make this error message more generic / better fitting for the Switchio case?

Reply:
Ah yes, I meant to change this, thanks for catching!

Reply:
Done in https://github.com/cal-itp/benefits/pull/2960/commits/a66716b8b1404a346b6d72739751abcefbf86bb2